keyboards/monsgeek/m2/m2.c

```diff
@@ -0,0 +1,205 @@
+    {0, D_9,   E_9,   F_9},
+    {0, D_10,  E_10,  F_10},
+    {0, D_11,  E_11,  F_11},
+    {0, D_12,  E_12,  F_12},
+    {0, D_13,  E_13,  F_13},
+    {0, D_14,  E_14,  F_14},
+    {1, D_8,    E_8,   F_8},
+    {1, D_9,    E_9,   F_9},
+    {1, D_10,  E_10,  F_10},
+    {1, G_7,   H_7,   I_7},
+
+    {0, G_1,   H_1,   I_1},
+    {0, G_2,   H_2,   I_2},
+    {0, G_3,   H_3,   I_3},
+    {0, G_4,   H_4,   I_4},
+    {0, G_5,   H_5,   I_5},
+    {0, G_6,   H_6,   I_6},
+    {0, G_7,   H_7,   I_7},
+    {0, G_8,   H_8,   I_8},
+    {0, G_9,   H_9,   I_9},
+    {0, G_10,  H_10,  I_10},
+    {0, G_11,  H_11,  I_11},
+    {0, G_12,  H_12,  I_12},
+    {0, G_13,  H_13,  I_13},
+    {1, G_4,   H_4,   I_4},
+    {1, G_5,   H_5,   I_5},
+    {1, G_6,   H_6,   I_6},
+
+    {0, J_1,   K_1,   L_1},
+    {0, J_2,   K_2,   L_2},
+    {0, J_3,   K_3,   L_3},
+    {0, J_4,   K_4,   L_4},
+    {0, J_5,   K_5,   L_5},
+    {0, J_6,   K_6,   L_6},
+    {0, J_7,   K_7,   L_7},
+    {0, J_8,   K_8,   L_8},
+    {0, J_9,   K_9,   L_9},
+    {0, J_10,  K_10,  L_10},
+    {0, J_11,  K_11,  L_11},
+    {0, J_12,  K_12,  L_12},
+    {1,  J_4,   K_4,  L_4},
+    {1,  J_7,   K_7,  L_7},
+    {1,  J_8,   K_8,  L_8},
+    {1,  J_9,   K_9,  L_9},
+    {1,  J_10,  K_10, L_10},
+
+    {0, J_13,  K_13,  L_13},
+    {0, J_14,  K_14,  L_14},
+    {0, J_15,  K_15,  L_15},
+    {0, J_16,  K_16,  L_16},
+    {0, G_14,  H_14,  I_14},
+    {0, G_15,  H_15,  I_15},
+    {0, G_16,  H_16,  I_16},
+    {1, J_2,  K_2,  L_2},
+    {1, J_3,  K_3,  L_3},
+    {1, J_5,  K_5,  L_5},
+    {1, J_6,  K_6,  L_6},
+};
+#endif //RGB_MATRIX_ENABLE
+
+enum __layers {
+    WIN_B,
+    WIN_WASD,
+    WIN_FN,
+    MAC_B,
+    MAC_WASD,
+    MAC_FN
+};
+void led_init_ports(void){
+    //set our LED pings as output
+    setPinOutput(LED_NUM_LOCK_PIN); // LED1 Num Lock
+    writePinLow(LED_NUM_LOCK_PIN);
+    setPinOutput(LED_CAPS_LOCK_PIN); // LDE2 Caps Lock
+    writePinLow(LED_CAPS_LOCK_PIN);
+    setPinOutput(LED_WIN_LOCK_PIN); // LED3 Win Lock
+    writePinLow(LED_WIN_LOCK_PIN);
+}
+
+bool led_update_kb(led_t led_state) {
+    bool res = led_update_user(led_state);
+    if(res) {
+        // writePin sets the pin high for 1 and low for 0.
+        // In this example the pins are inverted, setting
+        // it low/0 turns it on, and high/1 turns the LED off.
+        // This behavior depends on whether the LED is between the pin
+        // and VCC or the pin and GND.
+        writePin(LED_NUM_LOCK_PIN, led_state.num_lock);
+        writePin(LED_CAPS_LOCK_PIN, led_state.caps_lock);
+        writePin(LED_WIN_LOCK_PIN, keymap_config.no_gui);
+ }
+    return res;
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+        case DF(WIN_B):
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(WIN_B);
+                layer_state_set(1<<WIN_B);
```

Comment:
This shouldn't be needed.

If you're trying to clear active layers, then `layer_clear()` would be better.